contracts/NFTStorefrontV2.cdc

```diff
@@ -549,7 +549,12 @@ access(all) contract NFTStorefrontV2 {
     access(all) resource interface StorefrontPublic {
         access(all) view fun getListingIDs(): [UInt64]
         access(all) fun getDuplicateListingIDs(nftType: Type, nftID: UInt64, listingID: UInt64): [UInt64]
-        access(all) view fun borrowListing(listingResourceID: UInt64): &{ListingPublic}?
+        access(all) view fun borrowListing(listingResourceID: UInt64): &{ListingPublic}? {
+            post {
+                result == nil || result!.getType() == Type<@Listing>():
```

Comment:
The tradeoff here is composability which is fine, I'm just not sure if there are any actual implementations that leverage custom listing implementations. I think we could also prevent the same issue by adding a post-condition in the listing public interface's `purchase()`? Something like

```cadence
access(all) fun purchase(...): @{NonFungibleToken.NFT} {
  post {
    result.getType() == self.getDetails.nftType: "erorr message"
  }
  // ...
}
```

That would ensure that the you never purchase something you didn't intend to without sacrificing composability. Of course, resolving the listing would still be possible.

Reply:
I don't think this would work because you could still return an NFT with a different ID

Reply:
Composability with interfaces are unfortunately dead in my opinion, tbh with 1.0 there is not much need to use interfaces except FT, NFT and Metadata standards.  

To be fair no-one was using, I think interfaces, with default implementations, will be more like building blocks with 1.0

Reply:
Could we also add a second condition like below to cover for that?
```cadence
result.id == self.getDetails.nftID
```

Reply:
The listing would have the same ID as the result because it is a different listing that was returned when you called `borrowListing()`. I think this is the only way

Reply:
`getDetails` is also function there, not much to do.

Reply:
Got it, seems current implementation is the best solution. I'll approve, just wanted to explore if there was an option that was less prescriptive but seems not

Reply:
unfortunately yeah, trust modal is broken little. Especially with `@AnyResource{Interface}` syntax is now `@{Interface}` it is very easy to miss. ( e.g. `{NFTStorefrontV2.Listing}` vs `NFTStorefrontV2.Listing` when reading code )